zstd.go

```diff
@@ -8,23 +8,32 @@ import (
 
 // zstdMaxBufferedEncoders maximum number of not-in-use zstd encoders
 // If the pool of encoders is exhausted then new encoders will be created on the fly
-const zstdMaxBufferedEncoders = 1
+var zstdMaxBufferedEncoders int
 
 type ZstdEncoderParams struct {
-	Level int
+	Level               int
+	MaxBufferedEncoders int
 }
+
 type ZstdDecoderParams struct {
 }
 
 var zstdDecMap sync.Map
 
 var zstdAvailableEncoders sync.Map
 
+func getZstdMaxBufferedEncoders(params ZstdEncoderParams) int {
+	if params.MaxBufferedEncoders > 0 {
+		return params.MaxBufferedEncoders
+	}
+	return 1
+}
+
 func getZstdEncoderChannel(params ZstdEncoderParams) chan *zstd.Encoder {
-	if c, ok := zstdAvailableEncoders.Load(params); ok {
+	if c, ok := zstdAvailableEncoders.Load(params.Level); ok {
```

Comment:
🤔 Hm, not sure we want to clobber all `MaxBufferedEncoders` of the same `Level` together…

Maybe, we could do something like:
```
params.MaxBufferedEncoders = zstdMaxBufferedEncoders(params)
if c, ok := zstdAvailableEncoders.Load(params); ok {
…
```

Reply:
I think I probably should move MaxBufferedEncoders out of the ZstdEncoderParams struct.   The fields in ZstdEncoderParmszstdAvailableEncoders map was keyed on encoder parameters.  Encoders with the same encoder parameters are created to be the same kind of objects.  MaxBufferedEncoders are used to control the pool size, it's actually not part of the encoder parameter for creating an individual encoder object, use MaxBufferedEncoder as part of the map key would be confusing.

Reply:
OK, so the idea is that this would be set generically for all encoders in a process? And you just implemented that in a “stateless” sort of way, where everyone is expected to pass in the value?

Reply:
Yes, the 'maxBufferedEncoders' needs to pass in explicitly through function call.  ZstdEncoderParams is a struct for the parameters needed for creating an encoder.  maxBufferedEncoders is not a parameter for an individual encoder, it should not be stored in that param struct.